src/Umbraco.Community.Contentment/DataEditors/DataList/DataSources/JsonDataListSource.cs

```diff
@@ -3,16 +3,21 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
+using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Net;
+using System.Text;
 using Newtonsoft.Json.Linq;
-using Umbraco.Core;
-using Umbraco.Core.Composing;
 using Umbraco.Core.IO;
 using Umbraco.Core.Logging;
 using Umbraco.Core.PropertyEditors;
 
+// TODO
+// How would you get the string-value from a "key"?
+// This project https://github.com/s3u/JSONPath supports "~" to retrieve keys. However this is not in the original jsonpath-specs.
+// We could implement something similar, which checks the JsonPaths for a ~, and the we'll code-extract the keys. However this is a somewhat shady solution.
```

Comment:
I'm happy to stick with whatever JSON.NET supports. I hadn't heard about the "~" for keys - interesting!

Reply:
Yeah, but the ~would be a programmable extension where we need to analyse the jsonpath and act accordingly. A nice feature for the future, so I left it in the comments :)